### ISSUE TITLE
Enforce the correctness of Package objects

### ISSUE THREAD

**Issue by pombredanne** [nexB/scancode-toolkit#241]
The Package object holds quite a bit of data, and it is pretty loose right now in checking what is expected.
You can construct an incorrect one pretty easily.
We need some validation of sorts, possibly by defining some schema/validation library.
There are many such libraries so the hard part is picking a sensible and simple one.

**Comment by pombredanne** [comment]
Of these, http://awesome-python.com/#data-validation  this one http://schematics.readthedocs.org/en/latest/ seems the most promising at first glance.

**Comment by balusarakesh** [comment]
planning to use `http://docs.python-cerberus.org/en/stable/usage.html#validation-rules` as the validator.

**Comment by pombredanne** [comment]
@balusarakesh can you explain your rationale?
cerberus seems to be fine to validate arbitrary data, but does not seem to help define correct object models like schematics does. It has not been designed for that at all, whereas schematics has from what I can see

**Comment by balusarakesh** [comment]
When Model class (schematics package) is inherited into Package class it doesn't let us create new variables and throws an exception like 
```
raise ModelConversionError(errors, data)
          ModelConversionError: {'release': 'Rogue field', 'epoch': 'Rogue field', 'arch': 'Rogue field'}
```

**Comment by pombredanne** [comment]
So now that we have used schematics, I am not super happy with the way this works. In particular anything that would be stored in the DB in a near future would require having yet another schema being defined in yet another library. I made a few local tests storing things as raw JSON in a DB but that raise another problem of schema migrations.
I am considering using Django's ORM to define the models instead. I will try out some tests in a branch. The semantics are essentially the same as schematics or marshmallow (or the other way around) and there could be some side benefits to store an-flight scan in a temp sqlite DB. I will work out something.
Feedback welcomed
